src/tir/transforms/remove_weight_layout_rewrite_block.cc

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file remove_weight_layout_rewrite_block.cc
+ * \brief Remove weight layout rewrite block before benchmark
+ */
+
+#include <tvm/tir/op.h>
+#include <tvm/tir/stmt_functor.h>
+#include <tvm/tir/transform.h>
+
+namespace tvm {
+namespace tir {
+
+class WeightLayoutRewriteBlockRemover : public StmtMutator {
+ public:
+  static PrimFunc Remove(PrimFunc f) {
+    WeightLayoutRewriteBlockRemover remover;
+    auto* n = f.CopyOnWrite();
+    n->body = remover(std::move(n->body));
+    for (const auto& kv : f->buffer_map) {
+      const Var& param = kv.first;
+      const Buffer& buffer = kv.second;
+      auto it = remover.buf_map_.find(buffer);
+      if (it != remover.buf_map_.end()) {
+        n->buffer_map.Set(param, (*it).second);
+      }
+    }
+    return f;
+  }
```

Comment:
I'm a bit allergic to possible corner cases where we iterate a dict while updating it. Let's be more cautious here even if it's hypothetically not the case

```suggestion
  static PrimFunc Remove(PrimFunc f) {
    WeightLayoutRewriteBlockRemover remover;
    PrimFuncNode* n = f.CopyOnWrite();
    n->body = remover(std::move(n->body));
    Map<tir::Var, Buffer> buffer_map;
    for (const auto& kv : f->buffer_map) {
      Var param = kv.first;
      Buffer buffer = kv.second;
      auto it = remover.buf_map_.find(buffer);
      if (it != remover.buf_map_.end()) {
        buffer_map.Set(param, (*it).second);
      } else {
        buffer_map.Set(param, buffer);
      }
    }
    n->buffer_map = std::move(buffer_map);
    return f;
  }
```